scapy/contrib/gtp.py

```diff
@@ -679,27 +701,24 @@ def extract_padding(self, pkt):
 
 
 def IE_Dispatcher(s):
-  """Choose the correct Information Element class."""
-
-  if len(s) < 1:
-    return Raw(s)
-
-  # Get the IE type
-  ietype = ord(s[0])
-  cls = ietypecls.get(ietype, Raw)
-
-  # if ietype greater than 128 are TLVs
-  if cls == Raw and ietype & 128 == 128:
-    cls = IE_NotImplementedTLV
-
-  return cls(s)
+    """Choose the correct Information Element class."""
+    if len(s) < 1:
+        return Raw(s)
+    # Get the IE type
+    ietype = ord(s[0])
+    cls = ietypecls.get(ietype, Raw)
+
+    # if ietype greater than 128 are TLVs
+    if cls == Raw and ietype & 128 == 128:
+        cls = IE_NotImplementedTLV
+    return cls(s)
 
 class GTPEchoResponse(Packet):
     # 3GPP TS 29.060 V9.1.0 (2009-12)
     name = "GTP Echo Response"
     fields_desc = [ XBitField("seq", 0, 16),
                     ByteField("npdu", 0),
-                    ByteField("next_ex", 0),
+                    ByteEnumField("next_ex", 0, ExtensionHeadersTypes),

```

Comment:
same as above and for all the *Response and *Request classes, already in the GTPHeader